### DISCUSSION TITLE
Support NPs to be part of different networks

### DISCUSSION THREAD

**Discussion by NikithShetty** [beckn/protocol-specifications#345] [Ideas]
Beckn protocol is going to be adopted globally. Let's say a BAP or BPP wants to expand their business across geographies which have their own networks powered by beckn protocol, how can the same BAP/BPP stack validate and respond to requests coming from different networks?

*Proposed solution*
Every beckn API can have a network identifier, ex: `x-network-id: ondc.org`. Based on this header, every NP can have a mapping to the registry. This can help with validation of incoming requests.

**Comment by ravi-prakash-v** [comment]
The end goal is to have one large network with various policies layered on it. Let us not bring the concept of different networks in the protocol itself. Rather we can look at embedding the network identifier inside the subscriber ID. Like komn.example-bap.com where the subdomain can have a network identifier. Additionally there may need to be a root registry infrastructure that maintains identifiers of all networks.

**Reply by ravi-prakash-v** [reply]
It's like in IPV4, subnets can be defined within the same format using subnet masks.

**Reply by venkatramanm** [reply]
root registry, could just have entries for local registries. (by geography) Firing on a bg on root registry could cascade to bg on all other registries. there by mimicking something like the IP routing .eventually hitting relevant 
bpps.

**Comment by venkatramanm** [comment]
@NikithShetty , Every network has a registry with  subscriber_ids and possibly different subscriber URL. I have used different URLs in multiple networks for the usecase you are taking about. The URL could be /komn/bpp or /.ondc/bpp , Then the bpp can figure from the URL path which network it needs to check. In this solution, or the one @ravi-prakash-v is suggesting,  no one else needs to change. If you want to participate in multiple networks, you can by registering different URLs or different subscriber ids.. You may need to keep internal meta data by network to know the registry url etc. I have had reasonable success with the url story.

With respect to cross network or cross border transaction in a mega network, Is a great dream/goal to shoot for. As a guiding principle  it would sanitize the policies toward interoperability. .

In my opinion, the mega network would be composition of smaller networks and bridge platforms (bpp on one net while being bap on another) would handle network nuances for other NPS. (They may handle import/export customs, etc) 

Exciting possibilities . Don't know which of these evolutionary paths would be finalized by natural selection. :)

**Reply by NikithShetty** [reply]
Thanks @venkatramanm , this seems like a good solution since it doesn't require any protocol level changes. Good to know that you are using this solution as well 👍
